front/app/components/FormBuilder/edit/index.tsx

```diff
@@ -152,8 +159,19 @@ const FormEdit = ({
     }
   }, [formCustomFields, isUpdatingForm, isFetching, reset]);
 
-  const closeSettings = () => {
+  const closeSettings = async () => {
     setSelectedField(undefined);
+
+    if (autosaveEnabled) {
+      // Auto-save the form if the close button is clicked
+      const formValues = getValues();
+      handleSubmit(
+        onFormSubmit(formValues) as SubmitHandler<{
+          // Required casting here.
+          customFields: IFlatCustomField[];
+        }>
+      );
```

Comment:
what about just

```suggestion
      onFormSubmit(formValues);
```

Reply:
Oh - yes that works! I thought I tried that earlier and it didn't work since this is a React hook form, but indeed it saves correctly :+1: 

Thanks for pointing it out :raised_hands: